### ISSUE TITLE
WorkerConfig.fromFile fails when there is no pulsar config in the file

### ISSUE THREAD

**Issue by chrislatimer** [infiniticio/infinitic#126]
I'm trying to create a worker by using `PulsarInfiniticWorker.from(client, workerConfig)` where you supply a `PulsarClient`object and a `WorkerConfig` object. However, when I try to create the `WorkerConfig` using `WorkerConfig.fromFile(file)` it throws an exception because it expects the file to have a `pulsar` section. 

I'm working with the [example app here](https://github.com/infiniticio/infinitic-example-java-booking) using a Pulsar cluster that has authentication enabled and it isn't clear how to configure that in `infinitic.yml`.

Here's the exception:
```
- Could not instantiate 'io.infinitic.config.WorkerConfig' because:

        - 'pulsar': Missing from config
        at com.sksamuel.hoplite.ConfigLoader$returnOrThrow$1.invoke(ConfigLoader.kt:335)
        at com.sksamuel.hoplite.ConfigLoader$returnOrThrow$1.invoke(ConfigLoader.kt:20)
        at com.sksamuel.hoplite.fp.ValidatedKt.getOrElse(Validated.kt:93)
        at com.sksamuel.hoplite.ConfigLoader.returnOrThrow(ConfigLoader.kt:332)
        at io.infinitic.config.WorkerConfig$Companion.fromFile(WorkerConfig.kt:166)
        at io.infinitic.config.WorkerConfig.fromFile(WorkerConfig.kt)
        at example.booking.Worker.main(Worker.java:20)
```

If I add a pulsar section to the file (even though it gets ignored since I'm passing in an instantiated `PulsarClient`) the exception doesn't get throw.

**Comment by geomagilles** [comment]
@chrislatimer the pulsar configuration is mandatory to know which tenant and namespace to use. The  other parameters are optional

**Comment by chrislatimer** [comment]
My preference would have been to use the infinitic.yml file to configure the Pulsar connection, but it didn't seem to support authentication so I was trying to pass in the PulsarConfig object that was created using:

```
PulsarClient pclient = PulsarClient.builder()
                .serviceUrl(SERVICE_URL)
                .authentication(
                        AuthenticationFactory.token(jwt)
                )
                .build();
```

Is there a way to specify jwt authentication in the infinitic.yml file?

**Comment by geomagilles** [comment]
@chrislatimer not yet. For non-local configuration, you should use `PulsarInfiniticWorker.from(client, workerConfig)` for the moment. Note that you can create a `WorkerConfig` instance with `WorkerConfig.fromFile(...)` or `WorkerConfig.fromResource(...)`

I plan to include authentication in Infinitic.yml - but I need a user like you able to validate an implementation on an actual system.

**Comment by chrislatimer** [comment]
It looks like that would work as long as you add the `pulsar` section to the worker configuration file specifying the namespace and tenant. The syntax is a little inconsistent with the similar constructors for `PulsarInfiniticAdmin` and `PulsarInfiniticClient` which let you pass in the `PulsarAdmin` and `PulsarClient` objects respectively with String values for tenant/namespace but not that big of a deal.

**Comment by geomagilles** [comment]
I see what you mean. I see it as a transient situation, as you should be able to do everything with the configuration file eventually. The other syntax would be used for the rare unanticipated cases

**Comment by geomagilles** [comment]
this will be fixed by coming 0.8 version

**Comment by geomagilles** [comment]
I'm closing this - as you should now be able to connect with a configuration file such as:
```
pulsar:
  brokerServiceUrl: pulsar+ssl://pulsar-aws-useast2.streaming.datastax.com:6651
  webServiceUrl: https://pulsar-aws-useast2.api.streaming.datastax.com
  tenant: your_tenant
  namespace: your_namespace
  tlsEnableHostnameVerification: true
  authentication:
    token: your_token
```
